framework/include/utils/MathUtils.h

```diff
@@ -131,6 +131,35 @@ poly(std::vector<Real> c, const T x, const bool derivative)
   return value;
 }
 
+template <typename T>
+T
+clamp(T x, T lowerlimit, T upperlimit)
+{
+  if (x < lowerlimit)
+    return lowerlimit;
+  if (x > upperlimit)
+    return upperlimit;
+  return x;
+}
+
+template <typename T>
+T
+smootherStep(T x, T start, T end, bool derivative = false)
+{
+  x = clamp((x - start) / (end - start), 0.0, 1.0);
```

Comment:
You have a few potential division by zeros in here. I would recommend a `mooseAssert()` or some other explicit parameter check.

Reply:
Good point! I just returned 0.0 if start == end